spec/tapioca/dsl/compilers/rubocop_spec.rb

```diff
@@ -0,0 +1,149 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "spec_helper"
+require "rubocop"
+require "rubocop-sorbet"
+
+module Tapioca
+  module Dsl
+    module Compilers
+      class RuboCopSpec < ::DslSpec
+        EXISTING_CONSTANTS = Runtime::Reflection
+          .descendants_of(::RuboCop::Cop::Base)
+          .filter_map { |constant| Runtime::Reflection.name_of(constant) }
+
+        class << self
+          def target_class_file
+            # Against convention, RuboCop uses "rubocop" in its file names, so we do too.
+            super.gsub("rubo_cop", "rubocop")
+          end
```

Comment:
Originally, I'd renamed the files to `rubo_cop.rb` too, due to the loading not working, but I like this better, as it allows us to keep both the constant names and file names matching RuboCop's convention (which was the whole point of renaming the constants).

Reply:
Do we still need this since we don't rename the file in the DSL command?

Reply:
This allows us to have

- `lib/tapioca/dsl/compilers/rubocop.rb`
- `lib/tapioca/dsl/extensions/rubocop.rb`
- `spec/tapioca/dsl/compilers/rubocop_spec.rb`

instead of


- `lib/tapioca/dsl/compilers/rubo_cop.rb`
- `lib/tapioca/dsl/extensions/rubo_cop.rb`
- `spec/tapioca/dsl/compilers/rubo_cop_spec.rb`

which I think is preferable, especially given it's really just limited to telling the spec file how to require the correct compiler.

Reply:
I would rather rename the constant `Rubocop` or let the file be named `rubo_cop` rather than play with this.

It seems all the files under `dsl/compilers` follow the same convention. `class CompilerName` -> `compiler_name.rb`. I'm not sure why we should deviate from this.

Reply:
🤷 I was trying to match Rubocop's own naming convention (e.g. `RuboCop::Version` is defined in `lib/rubocop/version.rb`, not `lib/rubo_cop/version.rb`), but given the can of worms it would be to do this for the RBI files, maybe it's not worth doing for the compiler files either. It just feels weird seeing `rubo_cop` TBH, but I'm glad to deal with it if you think it's not worth the extra method in the test file.

Reply:
Okay, so circling back to this, https://github.com/Shopify/tapioca/commit/a80c4ecbd1215546c5c3e6fd470c55d30303a4ef makes the changes necessary to avoid this workaround, but my gut feeling is that this leads to the code being more confusing, because now we're dealing with a mix of constants that include `RuboCop` and constants that include `Rubocop`, and it's not obvious why they differ.

Likewise, we could go for the `rubo_cop` file naming, but then we'd have a mid of those files and ones like `.rubocop.yml`, again without an obvious reason why.

Therefore, I think it's overall more straightforward to keep the `RuboCop` naming to match upstream, and be consistent across our constants and theirs, and have this one workaround method with the comment documenting why it's there. I think this would be in line with the "principle of least surprise", in that this one method makes the rest of the app less surprising because things _just work_.